src/Model/batchprocessing/BatchProcessPyRad2CSV.py

```diff
@@ -0,0 +1,290 @@
+import os
+import platform
+from radiomics import featureextractor
+from src.Model.PatientDictContainer import PatientDictContainer
+from src.Model.batchprocessing.BatchProcess import BatchProcess
+import pandas as pd
+
+
+class BatchProcessPyRad2CSV(BatchProcess):
+    """
+    This class handles batch processing for the PyRadCSV process.
+    Inherits from the BatchProcess class.
+    """
+    # Allowed classes for PyRadCSV
+    allowed_classes = {
+        # RT Structure Set
+        "1.2.840.10008.5.1.4.1.1.481.3": {
+            "name": "rtss",
+            "sliceable": False
+        },
+        # RT Dose
+        "1.2.840.10008.5.1.4.1.1.481.2": {
+            "name": "rtdose",
+            "sliceable": False
+        },
+    }
+
+    def __init__(self, progress_callback, interrupt_flag, patient_files,
+                 output_path):
+        """
+        Class initialiser function.
+        :param progress_callback: A signal that receives the current
+                                  progress of the loading.
+        :param interrupt_flag: A threading.Event() object that tells the
+                               function to stop loading.
+        :param patient_files: List of patient files.
+        :param output_path: output of the resulting .csv file.
+        """
+        # Call the parent class
+        super(BatchProcessPyRad2CSV, self).__init__(progress_callback,
+                                                   interrupt_flag,
+                                                   patient_files)
+
+        # Set class variables
+        self.patient_dict_container = PatientDictContainer()
+        self.required_classes = 'rtss'.split()
+        self.ready = self.load_images(patient_files, self.required_classes)
+        self.output_path = output_path
+        self.filename = "Pyradiomics_.csv"
+
+    def start(self):
+        """
+        Goes through the steps of the PyRadCSV conversion.
+        """
+        # Stop loading
+        if self.interrupt_flag.is_set():
+            self.patient_dict_container.clear()
+            self.summary = "INTERRUPT"
+            return False
+
```

Comment:
If you can make summary messages enum or constants, and import them when you want to use that would be cleaner.